### ISSUE TITLE
[Accessibility]: Repository settings: Unusable on small viewports

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35923]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34428

### Problem description

<img width="516" alt="image" src="https://user-images.githubusercontent.com/9516420/169998712-e096510e-2f14-4869-b930-af1f456e9afe.png">

The sidebar takes up the vast majority of the space, and the content of the pages is very difficult to read.

### Expected behavior

Make the sidebar collapsible on small viewports.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
